docs/index.js

```diff
@@ -0,0 +1,621 @@
+    const amt = document.getElementById("tradeListAmount").value;
+    const listPayload = {
+      p: "krc-20",
+      op: "list",
+      tick: tick.toLowerCase(),
+      amt,
+    };
+    const scriptBuilder = createKRC20ScriptBuilder(listPayload);
+
+    const scriptPublicKey = scriptBuilder.createPayToScriptHashScript();
+    const P2SHAddress = kaspaWasm.addressFromScriptPublicKey(
+      scriptPublicKey,
+      "testnet-10",
+    );
+
+    const commitTxId = await kastle.signAndBroadcastTx("testnet-10", [
+      { amount: "0.3", address: P2SHAddress.toString() },
+    ]);
+    document.getElementById("P2SHListAddress").innerText =
+      P2SHAddress.toString();
+    document.getElementById("listCommitTxId").innerText = commitTxId;
+    document.getElementById("listScript").innerText = scriptBuilder.toString();
+
+    document.getElementById("listErrorKRC20").innerText = "";
+  } catch (error) {
+    document.getElementById("listErrorKRC20").innerText = error.message;
+  }
+});
+
+document.getElementById("krcListReveal").addEventListener("click", async () => {
+  const rpc = new kaspaWasm.RpcClient({
+    url: "wss://ws.tn10.kaspa.forbole.com/borsh",
+    networkId: "testnet-10",
+  });
+  await rpc.connect();
+
+  try {
+    const P2SHAddress = document.getElementById("P2SHListAddress").innerText;
+    const listScriptBuilder = kaspaWasm.ScriptBuilder.fromScript(
+      document.getElementById("listScript").innerText,
+    );
+
+    let P2SHEntries = [];
+    while (P2SHEntries.length === 0) {
+      const P2SHUTXOs = await rpc.getUtxosByAddresses([P2SHAddress.toString()]);
+      P2SHEntries = P2SHUTXOs.entries;
+
+      if (P2SHEntries.length === 0) {
+        await new Promise((resolve) => setTimeout(resolve, 1000));
+      }
+    }
+
+    const P2SHEntry = P2SHEntries[0];
+    const entry = {
+      address: P2SHEntry.address.toString(),
+      amount: kaspaWasm.sompiToKaspaString(P2SHEntry.amount),
+      scriptPublicKey: JSON.parse(P2SHEntry.scriptPublicKey.toString()),
+      blockDaaScore: P2SHEntry.blockDaaScore.toString(),
+      outpoint: JSON.parse(P2SHEntry.outpoint.toString()),
+    };
+
+    const sendPayload = {
+      p: "krc-20",
+      op: "send",
+      tick: document.getElementById("tradeListTick").value.toLowerCase(),
+    };
+
+    const sendScriptBuilder = createKRC20ScriptBuilder(sendPayload);
+    const sendScriptPublicKey = sendScriptBuilder.createPayToScriptHashScript();
+    const sendP2SHAddress = kaspaWasm.addressFromScriptPublicKey(
+      sendScriptPublicKey,
+      "testnet-10",
+    );
+
+    document.getElementById("sendP2SHAddress").innerText =
+      sendP2SHAddress.toString();
+    document.getElementById("sendScript").innerText =
+      sendScriptBuilder.toString();
+
+    const revealTxId = await kastle.signAndBroadcastTx(
+      "testnet-10",
+      [{ amount: "0.3", address: sendP2SHAddress.toString() }],
+      {
+        priorityEntries: [entry],
```

Comment:
Why priority entry instead of entry? To be sure it's positioned first?

Reply:
Yes, we need to make this utxo first position.

I also think that it would be better to use `signTx` instead of `signAndBroadcastTx` which generates transaction by `wasm.Generator` since `signTx` has more flexibility to adjust the content.